extensions/eda/plugins/event_source/pg_listener.py

```diff
@@ -0,0 +1,135 @@
+"""pg_listener.py.
+
+An ansible-rulebook event source plugin for reading events from
+pg_pub_sub
+
+Arguments:
+---------
+    dsn: The connection string/dsn for Postgres
```

Comment:
@Alex-Izquierdo That makes a tight coupling between our source plugin and the dependent package (psycopg, libpq), there are multiple formats for specifying the DSN and it will continue to evolve https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING-KEYWORD-VALUE

The psycopg does the validation and we don't have to do the same thing on our side. Otherwise we have to put the logic to parse or put back things in the correct order based on individual parameters. Leaving this to the psycopg will help us having a extensible solution.